core/src/main/scala/sttp/client/testing/SttpBackendStub.scala

```diff
@@ -86,8 +93,24 @@ class SttpBackendStub[F[_], S](
     }
   }
 
-  override def openWebsocket[T, WR](request: Request[T, S], handler: NothingT[WR]): F[WebSocketResponse[WR]] =
-    handler // nothing is everything
+  override def openWebsocket[T, WR](request: Request[T, S], handler: WS_HANDLER[WR]): F[WebSocketResponse[WR]] = {
+    Try(wsMatchers.lift(request)) match {
+      case Success(Some(openMatch)) =>
+        openMatch match {
+          case UseHandler(headers, useHandler) =>
+            val uz = useHandler.asInstanceOf[WS_HANDLER[WR] => WR]
+            monad.unit(WebSocketResponse(headers, uz(handler)))
+          case ReturnWebsocketResponse(headers, response) =>
+            monad.map(response) { r => WebSocketResponse(headers, r.asInstanceOf[WR]) }
+        }
+      case Success(None) =>
+        fallback match {
+          case None     => monad.error(new ReadException(new Exception("request not matched any stub path")))
+          case Some(fb) => fb.openWebsocket(request, handler)
+        }
+      case Failure(e) => monad.error(e)
```

Comment:
Is it possible to flatten it this way?
```
case Success(Some(UseHandler(headers, useHandler))) =>
  val uz = useHandler.asInstanceOf[WS_HANDLER[WR] => WR]
  monad.unit(WebSocketResponse(headers, uz(handler)))
case Success(Some(ReturnWebsocketResponse(headers, response))) =>
  monad.map(response) { r => WebSocketResponse(headers, r.asInstanceOf[WR]) }
case Success(None) =>
  fallback match {
    case None     => monad.error(new ReadException(new Exception("request not matched any stub path")))
    case Some(fb) => fb.openWebsocket(request, handler)
  }
case Failure(e) => monad.error(e)
```

Reply:
Will do! Also this `uz` was meant to be temporary name.